addons/hr/models/hr_employee.py

```diff
@@ -159,14 +159,13 @@ def _compute_avatar_128(self):
         super()._compute_avatar_128()
 
     def _compute_avatar(self, avatar_field, image_field):
-        for employee in self:
+        user_image_present_employees = self.filtered(lambda employee: employee.user_id or employee._origin[image_field])
+        for employee in user_image_present_employees:
```

Comment:
`filtered` loops on `self`, then we loop again. To avoid that:
```suggestion
        employee_wo_user_or_image_ids = []
        for employee in self:
            if not (employee.user_id or employee._origin[image_field]):
                employee_wo_user_or_image_ids.append(employee.id)
                continue
            ...
        super(HrEmployeePrivate, self.browse(employee_wo_user_or_image_ids))._compute_avatar(avatar_field, image_field)
```
Note that I create list of ids, in order not to browse each time I append. :slightly_smiling_face:

Reply:
@auon-odoo,
Please check the changes made in the commit whenever you have time.
Thanks!!